package-res/resources/web/pentaho/type/Instance.js

```diff
@@ -141,7 +141,8 @@ define([
       if(!keyArgs) keyArgs = {};
 
       var scope = new SpecificationScope();
-      var requireType = !keyArgs.omitRootType;
+      //anonymous types should always return their `_` but the omitRootType flag may override this behavior
+      var requireType = (!keyArgs.omitRootType ? (!this.id ? true : false) : false);
```

Comment:
You requireType independently of it having an id. If type does not have an id, it will have to be serialized using a generic type spec... (not just an id, string reference)